pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerTest.java

```diff
@@ -100,6 +100,12 @@ private Object[][] provideTestSqlAndRowCount() {
         new Object[]{"SELECT a.col1, a.ts, b.col2, b.col3 FROM a JOIN b ON a.col1 = b.col2 "
             + " WHERE a.col3 >= 0 AND a.col2 = 'alice' AND b.col3 >= 0", 3},
 
+        // Join query with IN and Not-IN clause. Table A's side of join will return 9 rows and Table B's side will
+        // return 2 rows. Join will be only on col1=bar and since A will return 3 rows with that value and B will return
+        // 1 row, the final output will have 3 rows.
+        new Object[]{"SELECT a.col1, b.col2 FROM a JOIN b ON a.col1 = b.col1 "
```

Comment:
can we add more test. `IN` and `NOT IN` with 0, 1, and more than 1 values

Reply:
also add test for implicit range search, such as 
- `WHERE (a.col3 > 0 AND a.col3 < 10) OR a.col3 >= 30`
- `WHERE a.col2 = 'foo' OR a.col2 = 'bar'`

Reply:
I can add the range search test in the range PR that I'll raise later this week?

This PR only focuses on IN/Not-In. Ranges won't work yet.

Reply:
Updated. Query fails parsing if there are no arguments to the IN clause, i.e. `col1 IN ()` fails